### PR TITLE
TWI hevy staunch martyr

### PR DESCRIPTION
**Pull request by ssoulier**
Adding [Hevy](https://swudb.com/card/TWI/164/hevy).

Screenshot 2024-12-11 at 16 01 35

I lost time understanding that I should not use targetResolver. I am still learning :)

### CODE REVIEW

File: test/server/cards/03_TWI/units/HevyStaunchMartyr.spec.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                        groundArena: ['specforce-soldier', 'luke-skywalker#jedi-knight', 'hevy#staunch-martyr'],
+                    },
+                    player2: {
+                        groundArena: ['consular-security-force', 'wampa', 'fleet-lieutenant'],
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.hevy);
```
Comment (VJubert) [inline test/server/cards/03_TWI/units/HevyStaunchMartyr.spec.ts:18]: can you check power before and after coordinate is active ?

File: server/game/cards/03_TWI/units/HevyStaunchMartyr.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            type: AbilityType.Constant,
+            title: 'Gain Raid 2',
+            ongoingEffect: AbilityHelper.ongoingEffects.gainKeyword({ keyword: KeywordName.Raid, amount: 2 }),
+        });
+
+        this.addWhenDefeatedAbility({
+            title: 'Deal 1 damage to each enemy ground unit.',
+            immediateEffect: AbilityHelper.immediateEffects.damage((context) => ({
+                amount: 1,
+                target: context.source.controller.opponent.getUnitsInPlay(ZoneName.GroundArena)
```
Comment (VJubert) [inline server/game/cards/03_TWI/units/HevyStaunchMartyr.ts:24]: I think unitInPlay should be replace by getArenaUnits

**Review by VJubert — APPROVED** [review]

File: test/server/cards/03_TWI/units/HevyStaunchMartyr.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+                // When defeated, deal 1 damage to each enemy ground unit
+                context.player2.clickCard(context.wampa);
+                context.player2.clickCard(context.hevy);
+                expect(context.hevy).toBeInZone('discard');
+                expect(context.wampa).toBeInZone('discard');
+                expect(context.consularSecurityForce).toBeInZone('discard');
+                expect(context.fleetLieutenant.damage).toBe(1);
+                expect(context.miningGuildTieFighter.damage).toBe(0);
+                expect(context.miningGuildTieFighter).toBeInZone('spaceArena');
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/units/HevyStaunchMartyr.spec.ts:32]: We should also check the damage on friendly ground units to make sure the ability is targeting correctly

File: test/server/cards/03_TWI/units/HevyStaunchMartyr.spec.ts (hunk 1 above)
Comment (ssoulier) [reply test/server/cards/03_TWI/units/HevyStaunchMartyr.spec.ts:19]: I did something. Not 100% it's what you had in mind. Let me know.

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**